sky/cli.py

```diff
@@ -342,6 +343,48 @@ def _parse_env_var(env_var: str) -> Tuple[str, str]:
 ]
 
 
+def _complete_cluster_name(ctx: click.Context, param: click.Parameter,
+                           incomplete: str) -> List[str]:
+    """Handle shell completion for cluster names."""
+    del ctx, param  # Unused.
+    return global_user_state.get_cluster_names(incomplete)
+
+
+def _install_shell_completion(ctx: click.Context, param: click.Parameter,
+                              value: str):
+    """A callback for installing shell completion for click."""
+    del param  # Unused.
+    if not value or ctx.resilient_parsing:
+        return
+
+    if value == 'bash':
+        cmd = '_SKY_COMPLETE=bash_source sky > ~/.sky-complete.bash && \
+                echo "# For SkyPilot shell completion" >> ~/.bashrc && \
+                echo ". ~/.sky-complete.bash\n" >> ~/.bashrc'
+
+    elif value == 'fish':
+        cmd = '_SKY_COMPLETE=fish_source sky > \
+                ~/.config/fish/completions/sky.fish'
+
+    elif value == 'zsh':
+        cmd = '_SKY_COMPLETE=zsh_source sky > ~/.sky-complete.zsh && \
+                echo "# For SkyPilot shell completion" >> ~/.zshrc && \
+                echo ". ~/.sky-complete.zsh\n" >> ~/.zshrc'
+
+    else:
+        click.secho(f'Unsupported shell: {value}', fg='yellow')
+        ctx.exit()
+
+    try:
+        subprocess.run(cmd, shell=True, check=True)
+        click.secho(f'Shell completion installed for {value}', fg='green')
+        click.echo('Completion will take effect once you restart the terminal')
```

Comment:
UX: Can we add the actual commands for restarting the terminal as a hint for the user, e.g. `source ~/.bashrc` or `exec bash`?